### PR TITLE
Dereference pointers when comparing MaintainerCanModify

### PR DESCRIPTION
**Pull request by msal4**
#### Summary
Right now we're comparing pointers which is probably not the behavior we want

```go
// server/pull_request_handler.go

//...
if oldPr.MaintainerCanModify == nil || (oldPr.MaintainerCanModify != pr.MaintainerCanModify) {
	prHasChanges = true
}
//...
```
I was gonna commit the change in #257 but I wasn't quick enough :)
#### Ticket Link
JIRA: https://mattermost.atlassian.net/browse/MM-26814

### CODE REVIEW

**Review by agnivade — CHANGES_REQUESTED** [review]
Woops, nice catch! Can we write a test for this please so that the behavior does not regress again?

**Comment by msal4** [comment]
I think these should do it.
I'll do a separate PR after this one is merged to update `Merged, MaintainerCanModify, MilestoneNumber, MilestoneTitle` to be `*type ` since the changes are related and minuscule.

File: server/pull_request_handler_test.go [hunk 1]
```diff
@@ -245,6 +245,168 @@ func TestPullRequestEventHandler(t *testing.T) {
...
+					Name:  &modelPR.RepoName,
+				},
+			},
+			Merged: &modelPR.Merged.Bool,
+			Head:   &github.PullRequestBranch{SHA: &modelPR.Sha},
+			Milestone: &github.Milestone{
+				Number: NewInt(int(modelPR.MilestoneNumber.Int64)),
+				Title:  &modelPR.MilestoneTitle.String,
+			},
+			MaintainerCanModify: NewBool(false),
```
Comment (agnivade) [inline server/pull_request_handler_test.go:342]: I'd prefer to do a `!` here too to be consistent and more robust.

**Review by agnivade — APPROVED** [review]

**Review by isacikgoz — APPROVED** [review]
LGTM 👍  Thanks for the unit tests.

**Pull request merged**